src/Cli/dotnet/SudoEnvironmentDirectoryOverride.cs

```diff
@@ -33,7 +37,57 @@ public static void OverrideEnvironmentVariableToTmp(ParseResult parseResult)
             if (!OperatingSystem.IsWindows() && IsRunningUnderSudo() && IsRunningWorkloadCommand(parseResult))
             {
                 Directory.CreateDirectory(SudoHomeDirectory);
+
+                var homeBeforeOverride = Path.Combine(Environment.GetEnvironmentVariable("HOME"));
```

Comment:
Sorry, I must have missed this in the first check in of this change, but directly using the HOME env var doesn't match the logic we use for calculating the home dir here: https://github.com/dotnet/sdk/blob/release/6.0.1xx/src/Common/CliFolderPathCalculatorCore.cs#L35, should we be taking those other env vars into account?

Reply:
That's fine. If the user override the home directory, they are in control of the situation. We just need to make sure the default make sense.